### ISSUE TITLE
tracing using puppeteer produces very different results than manual timelines

### ISSUE THREAD

**Issue by catdad** [puppeteer/puppeteer#3913]
I am trying to automate some test scenarios with timeline results, and am noticing vastly different results in my timeline output when I perform an action with puppeteer vs. manually. This will be easier to explain with some images.

Here is the timeline for an action using puppeteer:

![image](https://user-images.githubusercontent.com/2205537/52290628-71795900-293e-11e9-9590-6bb35d144d06.png)

Here is the timeline for performing the same action manually:

![image](https://user-images.githubusercontent.com/2205537/52290658-8b1aa080-293e-11e9-8eb6-491bd0eeaa64.png)

Notice how in the manual case, javascript execution is not constant. It has dips, there are breaks between blocking code blocks, etc. In the puppeteer case, it is one solid execution block for the entire duration. I was able to reproduce this on multiple runs (both the puppeteer and manual cases).

In the manual case, I launched chrome using all the same flags that puppeteer uses:

```
"C:\Program Files (x86)\Google\Chrome\Application\chrome.exe" --disable-background-networking --enable-features=NetworkService,NetworkServiceInProcess --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-breakpad --disable-client-side-phishing-detection --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=site-per-process,TranslateUI --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --disable-sync --force-color-profile=srgb --metrics-recording-only --no-first-run --safebrowsing-disable-auto-update --enable-automation --password-store=basic --use-mock-keychain about:blank --user-data-dir="C:\Users\kvatev\AppData\Local\Temp\puppeteer_dev_profile-BBX03U"
```

In the case for puppeteer, I used the same chrome executable path, in order to keep the test case consistent.

The actual app being tested is currently under development, though if necessary, I could try to provide the a URL for it. The test case is very simple and quite easy to reproduce (it takes a single keyboard command to trigger is, and does not require a ton of user input that could differ between a manual case and an automated case). This is a React application, and performs a lot of asynchronous tasks. However, all the code is on the client-side and does not fetch anything from a server for this test case.

I am wondering, is there anything that puppeteer does in order to cause javascript execution to not pause, and if so, can I turn it off.

**Tell us about your environment:**

* Puppeteer version: 1.12.1
* Platform / OS version: Windows 10
* Node.js version: 10.10.0

**Comment by catdad** [comment]
For posterity, I added a new test case. In this case, I am using puppeteer to launch a browser, and then I have a 2 minute timeout. During that time, I perform all actions manually myself in the browser that puppeteer opened. I am seeing the same behavior, where the javascript execution happens in one single large block instead of multiple smaller blocks:

![image](https://user-images.githubusercontent.com/2205537/52291552-9d95d980-2940-11e9-81b2-735e3f2afe43.png)

**Comment by aslushnikov** [comment]
> I am wondering, is there anything that puppeteer does in order to cause javascript execution to not pause, and if so, can I turn it off.

@catdad interesting, I don't think we do anything that should cause this behavior. Any chance you can share something so that we can reproduce this?

**Comment by catdad** [comment]
@aslushnikov I put together a quick repro using the public demo of our product. I ran this just now to validate that it reproduces the problem, though note that we may change the demo in the future (sorry, I don't control that part).

https://github.com/catdad-experiments/puppeteer-tracing-bug-repro

The repro steps for manually testing are pretty simply... go to the URL in that repo, open the "blank" document from the splash page, click in the middle of the first page and paste the first 3 chapters of War and Peace ([you can get that from project gutenberg](https://www.gutenberg.org/files/2600/2600-h/2600-h.htm)).

If I have time later, I might try to create a simple webpage that also shows this issue, so that it is not using such a complicated web app.

**Comment by catdad** [comment]
@aslushnikov Thank you for looking at this bug!

I finally had some time to look deeper into this, and it may come to no surprise that this confusing was caused by user error. I had an issue with my automated case which did differed from the manual case, causing a different code path in my own app's code. After I fixed that issue in my test case, I was able to generate automated timelines that reproduced the behavior of the manual case (and vice versa). There was no fault in puppeteer.
